samples/client/petstore/elixir/test/format_test.exs

```diff
@@ -21,20 +21,21 @@ defmodule FormatTest do
              pattern_with_digits: "1234567890",
              pattern_with_digits_and_delimiter: "Image_01"
            }
-           |> FormatTest.decode() ==
+           |> then(fn params -> FormatTest.changeset(%FormatTest{}, params) end)
+           |> Ecto.Changeset.apply_action!(:insert) ==
              %FormatTest{
                integer: 1,
                int32: 2,
                int64: 3,
                number: 4.1,
                float: 5.2,
                double: 6.3,
-               decimal: "7.4",
+               decimal: 7.4,
```

Comment:
The previous version did not handle type-casting of decimal values. This is a nice goodie we get from switching to Ecto.